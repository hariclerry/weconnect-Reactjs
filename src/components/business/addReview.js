import React from "react";

import Notifications from "react-notify-toast";

import Footer from "components/common/footer";
import NavBar from "components/common/navBar";
import "static/css/bootstrap.min.css";
import "static/css/box-model.css";
import "static/css/fix-footer.css";

export const ReviewBusiness = ({ reviews, addReview, businessName }) => (
  <div>

     <Notifications />
    <NavBar />

    <div className="container">
      <div style={{ margin: "20px" }}>
      <div className="col-md-8 col-md-offset-2" className="box">
        <h1 >{businessName.name}</h1>
        <p>
          <b>Category: </b> {businessName.category}
        </p>
        <p>
          <b>Location: </b>
          {businessName.location}
        </p>
        <p>
          <b>Description: </b>
          {businessName.description}
        </p>

      </div>
      </div>

      <hr />
    </div>

    <div style={{margin: "50px 150px" }}>
      <form onSubmit={addReview}>
        <div className="form-group">
          <label className="control-label " htmlFor="description">
            Say Something about this Business:
          </label>
          <textarea
            className="form-control"
            cols="2"
            id="description"
            name="description"
            rows="2"
            required
          />
        </div>

        <div className="form-group">
          <button className="btn btn-primary " name="submit" type="submit">
            Add Review
          </button>
        </div>
      </form>
    </div>


       <div style={{ width: "auto", margin: "50px 150px"}}>
        <h3>Reviews</h3>
        { !reviews ? <div> No Reviews found </div>
        : 
        reviews.map((review, index) => (
        <p key={review.id}>
                
          <div style={{ width: "auto", padding: "10px", border: "0 solid gray", margin: "0", backgroundColor: "#e6e6e6"}}>             
          <ul className="list-group">
        <h4 style={{color: "green"}}>{review.username}:</h4>
        {review.description}
        </ul>
        </div>
        </p>
                  ))
                }
      
      </div>


    <Footer />
  </div>
);

export default ReviewBusiness;
