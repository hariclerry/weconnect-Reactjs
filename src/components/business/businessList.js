import React from "react";
import { Link } from "react-router-dom";

import BusinessLists from "components/business/displayBusinessList";
import SearchBusiness from "components/business/searchBar";
import NavBar from "components/common/navBar";
import Footer from "components/common/footer";
import "static/css/bootstrap.min.css";

const Businesses = ({businesses, searching, searchedBusinesses,
  searchBusinesses
  }) => {
  let searchedBusinessesList;
  if (searchedBusinesses == null) {
    searchedBusinessesList = [];
  } else {
    searchedBusinessesList = searchedBusinesses;
  }
  return (
    <div>
      <NavBar />

      <div style={{ float: "right", width: "50%" }}>
        <SearchBusiness searchBusinesses={searchBusinesses} />
      </div>

      <div className="container">
        <div style={{ margin: "100px" }}>
          <div>
            <h4>Below are a list of all Registered businesses</h4>
          </div>
        </div>
        <div className="table-responsive">
          <table className="table table-striped">
            <thead>
              <tr>
                <td>
                  <b>Name</b>
                </td>
                <td>
                  <b>Category</b>
                </td>
                <td>
                  <b>Location</b>
                </td>
                <td>
                  <b>Description</b>
                </td>
                <td>
                  <b>Review</b>
                </td>
              </tr>
            </thead>
            <tbody>
              {searching
                ? searchedBusinessesList.map((business, index) => (
                   <BusinessLists business={business}/> 
                  ))
                : businesses.map((business, index) => (
                  <BusinessLists business={business}/> 
                    
                  ))}
            </tbody>
          </table>
        </div>
      </div>

      <Footer />
    </div>
  );
};

export default Businesses;
