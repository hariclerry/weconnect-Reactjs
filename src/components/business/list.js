import React from "react";

import BusinessLists from "components/business/results";
import SearchBusiness from "components/business/searchBar";
import NavBar from "components/common/navBar";
import Footer from "components/common/footer";
import TableSet from "../common/tableSet";
import "static/css/bootstrap.min.css";


const Businesses = ({businesses, searching, searchedBusinesses,
  searchBusinesses
  }) => {
  let searchedBusinessesList;
  if (searchedBusinesses == null) {
    searchedBusinessesList = [];
  } else {
    searchedBusinessesList = searchedBusinesses;
  }
  return (
    <div>
      <NavBar />

      <div style={{ float: "right", width: "50%" }}>
        <SearchBusiness searchBusinesses={searchBusinesses} />
      </div>

      <div className="container">
        <div style={{ margin: "100px" }}>
          <div>
            <h4>Below are a list of all Registered businesses</h4>
          </div>
        </div>
        <div className="table-responsive">
          <table className="table table-striped">
            <TableSet />
            <tbody>
              {searching
                ? searchedBusinessesList.map((business, index) => (
                   <BusinessLists business={business}/> 
                  ))
                : businesses.map((business, index) => (
                  <BusinessLists business={business}/> 
                    
                  ))}
            </tbody>
          </table>
        </div>
      </div>

      <Footer />
    </div>
  );
};

export default Businesses;
