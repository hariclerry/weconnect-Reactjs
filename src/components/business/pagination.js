import React, { Component } from "react";


const Pagination = ({ prevPage, nextPage, paginate }) => {
  let paginateLinks = [
    { name: "prev", page: prevPage },
    { name: "next", page: nextPage }
  ];

  return (
    <nav aria-label="Page navigation example" style={{ margin: "0 500px" }}>
      <ul className="pagination justify-content-center">
        {paginateLinks.map(obj => {
          return obj.page === null ? (
            <li key={obj.name} className="page-item">
              <button
                className="page-link"
                href=""
                name={obj.name + "Page"}
                style={{ color: "#E6E6E6" }}
                disabled
              >
                {obj.name}
              </button>
            </li>
          ) : (
            <li key={obj.name} className="page-item">
              <button
                className="page-link"
                href=""
                name={obj.name + "Page"}
                onClick={() => paginate(parseInt(obj.page, 10))}
              >
                {obj.name}
              </button>
            </li>
          );
        })}
      </ul>
    </nav>
  );
};
export default Pagination;
