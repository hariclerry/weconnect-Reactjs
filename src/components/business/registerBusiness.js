import React from "react";

import Notifications from "react-notify-toast";

import Footer from "components/common/footer";
import NavBar from "components/common/navBar";
import "static/css/bootstrap.min.css";

const RegisterBusiness = ({ createBusiness }) => (
  <div>
    <Notifications />
    <NavBar />

    <div className="container">
      <h3> You can now Register your Business</h3>
    </div>
    <hr />
    <div style={{ margin: "100px" }}>
      <form onSubmit={createBusiness}>
        <div className="form-group">
          <label className="control-label requiredField" htmlFor="title">
            Business Name:
          </label>
          <input
            className="form-control"
            name="name"
            id="title"
            type="text"
            required
          />
        </div>

        <div className="form-group">
          <label className="control-label " htmlFor="category">
            Business Category:
          </label>
          <input
            className="form-control"
            id="category"
            name="category"
            type="text"
            required
          />
        </div>

        <div className="form-group">
          <label className="control-label " htmlFor="location">
            Business Location:
          </label>
          <input
            className="form-control"
            id="location"
            name="location"
            type="text"
            required
          />
        </div>

        <div className="form-group">
          <label className="control-label " htmlFor="description">
            Business Description:
          </label>
          <textarea
            className="form-control"
            cols="40"
            id="description"
            name="description"
            rows="10"
            required
          />
        </div>

        <div className="form-group">
          <button className="btn btn-primary " name="submit" type="submit">
            Create
          </button>
        </div>
      </form>
    </div>
    <Footer />
  </div>
);

export default RegisterBusiness;
