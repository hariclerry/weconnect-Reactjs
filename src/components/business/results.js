import React from "react";
import { Link } from "react-router-dom";

import "static/css/bootstrap.min.css";

const BusinessLists = ({ business }) => {
  
  return (
    <tr key={business.id}>
      <td>{business.name}</td>
      <td>{business.category}</td>
      <td>{business.location}</td>
      <td>{business.description}</td>
      <td>
        <Link
          to={`/addreview/${business.id}`}
          className="btn btn-info"
          role="button"
          style={{ marginRight: "20px" }}
        >
          Review
        </Link>
      </td>
    </tr>
  );
};

export default BusinessLists;
