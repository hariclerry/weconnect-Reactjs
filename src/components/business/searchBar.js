import React from "react";

import "static/css/bootstrap.min.css";

const SearchBusiness = ({ searchBusinesses }) => {
  return (
    <div>
      <form
        onSubmit={e => {
          e.preventDefault();
          searchBusinesses(e.target.elements.name.value);
        }}
        method="POST"
      >
        <div>
          <div className="col-md-8 col-md-offset-2 businesssearch">
            <div className="input-group">
              <input
                type="text"
                className="form-control"
                placeholder="Search business"
                name="name"
              />
              <span className="input-group-btn">
                <button className="btn btn-info" type="submit" id="search">
                  <i className="fa fa-search fa-fw" /> Search
                </button>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  );
};

export default SearchBusiness;
