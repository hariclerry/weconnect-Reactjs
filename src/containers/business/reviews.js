import React from "react";

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import {notify} from "react-notify-toast";

import { jsonStringify } from "utils/jsonHelper";
import AddReview from "components/business/addReview";
import { singleBusiness, fetchReview, reviewBusiness } from "actions/businessActions";
import { checkIfUserIsLoggedIn } from "actions/userActions";

/**
 * Review Container Component for adding a review and view all the reviews.
 * 
 */

class Reviews extends React.Component {

  componentDidMount() {
    checkIfUserIsLoggedIn(this.props.email, this.props.history);

    //  execute single business function and returns business information 
    const { id } = this.props.match.params;
    this.props.singleBusiness(id);

    //  execute fetch review function
    const businessId = this.props.match.params.id;
    this.props.fetchReview(businessId);
  }

  addReview = e => {
    // Prevent default submit event trigger(synthetic event):
    e.preventDefault(); 
    let reviewCredential = {
      description: e.target.elements.description.value
    };
    const businessId = this.props.match.params.id;
    this.props.reviewBusiness(businessId, jsonStringify(reviewCredential))
    .then(() => {
      notify.show("Reviews Successfully Added", "success");
    })
    
    window.location.reload();
  };

  render() {
    const reviews = Object.values({ ...this.props.fetchReview });

    if (reviews) {
      Array.prototype.reverse.call(reviews);
    }

    return (
      <AddReview
        reviews={this.props.reviews}
        addReview={this.addReview}
        username={this.props.username}
        businessName={this.props.businessName}
      />
    );
  }
}
Reviews.propTypes = {
  email: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  businessReviews: PropTypes.array.isRequired,
  fetchReview: PropTypes.func.isRequired,
  businessName: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  console.log(state);
  return {
    username: state.auth.loginData.username,
    email: state.auth.loginData.email,
    businessReviews: state.business.fetchReviewMessage,
    reviews: state.business.fetchReviewMessage.review_data,
    businessName: state.business.singleBusinessMessage.business_data,
  };
};

const mapDispatchToProps = dispatch => ({
  singleBusiness: id => dispatch(singleBusiness(id)),
  fetchReview: id => dispatch(fetchReview(id)),
  reviewBusiness: (id, data) => dispatch(reviewBusiness(id, data))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Reviews)
);
