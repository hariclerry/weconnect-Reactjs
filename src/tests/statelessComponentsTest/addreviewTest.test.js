import React from "react";
import {shallow} from "enzyme";
import { Reviews } from "../../containers/business/reviews"



const promiseFn = () =>  new Promise((resolve, reject) =>{
  
});

describe('reviewBusiness', ()=>{
  let wrapper;

  beforeEach(()=>{
    const history = {
      push: jest.fn
    }

  const  match = {
      params:  1
    }
  

    wrapper = shallow(<Reviews reviewBusiness={promiseFn} history={history} match={match} singleBusiness={promiseFn} fetchReview={promiseFn}/>)
  })

  it('should review business', ()=>{
    const e = {
      preventDefault: jest.fn,
      target:{
        elements:{
          description: {
            value: "This is andela"
          }
        }
      }
    }
    
    const spy = jest.spyOn(wrapper.instance(), "addReview");
    expect(spy).toHaveBeenCalledTimes(0);
    wrapper.instance().addReview(e);
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalledTimes(1);
  })

  it('should match snapshot', () =>{
    expect(wrapper).toMatchSnapshot();
  })
})
