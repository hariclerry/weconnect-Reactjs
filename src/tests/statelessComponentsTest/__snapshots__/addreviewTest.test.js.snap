// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reviewBusiness should match snapshot 1`] = `
<Component
  addReview={[Function]}
/>
`;
